Bottom: 5ae93ebe078d69d06cc619f628800c0f076b819f
Top:    1c7098bdbd9de5567090cd870f26b6f516e9d04f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 22:00:52 +0000

Refresh of drop-in-pcb-gl-code-various-me

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index ec0018a..a4632ee 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1019,47 +1019,40 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
              gport->offlimits_color.green / 65535.,
              gport->offlimits_color.blue / 65535.);
 
+  glBegin (GL_QUADS);
   if (eleft > 0)
     {
-      glBegin (GL_QUADS);
       glVertex2i (0, 0);
       glVertex2i (eleft, 0);
       glVertex2i (eleft, gport->height);
       glVertex2i (0, gport->height);
-      glEnd ();
     }
   else
     eleft = 0;
-
   if (eright < gport->width)
     {
-      glBegin (GL_QUADS);
       glVertex2i (eright, 0);
       glVertex2i (gport->width, 0);
       glVertex2i (gport->width, gport->height);
-      glVertex2i (eright, gport->width);
-      glEnd ();
+      glVertex2i (eright, gport->height);
     }
   else
     eright = gport->width;
   if (etop > 0)
     {
-      glBegin (GL_QUADS);
       glVertex2i (eleft, 0);
       glVertex2i (eright, 0);
       glVertex2i (eright, etop);
       glVertex2i (eleft, etop);
-      glEnd ();
     }
   if (ebottom < gport->height)
     {
-      glBegin (GL_QUADS);
       glVertex2i (eleft, ebottom);
       glVertex2i (eright + 1, ebottom);
       glVertex2i (eright + 1, gport->height);
       glVertex2i (eleft, gport->height);
-      glEnd ();
     }
+  glEnd ();
 
   /* TODO: Background image */
